enhancements/operator-framework-api/catalogue-check.md

```diff
@@ -0,0 +1,377 @@
+
+```sh
+$ operator-sdk bundle validate --list-optional
+NAME           LABELS                     DESCRIPTION
+operatorhub    suite=operatorframework    OperatorHub.io metadata validation
+               name=operatorhub          
+```
+
+**Cons**
+
+- Provide specific OpenShift solutions in upstream might hurt the community
+- OLM team would not like to add a code which is valid only for OpenShift unless we provide others vendor's validators as well in the public API. They are trying decouple the rules which are specific for OCP from OLM.
+- In the future it might result an effort required to move any OpenShift rules for downstream repos and implementations which are only valid for it. Currently, it would be the only case scenario so far, however, the precedence can bring others ones.
+
+### (Option B) Include the OpenShift Validator in downstream component for [operator-framework/api][oper-api] and use it in SDK upstream repository
+
+**Pros**
+
+- Mainly all pros of option A
+- Solves the OLM team concern about add OpenShift checks in [operator-framework/api][oper-api]
+  
+**Cons**
+
+- Effort required to keep a downstream repository for a downstream component and its releases [operator-framework/api][oper-api]
+- SDK would need to probably use only the downstream import to avoid misleading
+- It could cause delays for the SDK be able to apply the required changes of [operator-framework/api][oper-api] which can result ,for example, in an impact for Pipeline and users to get bug fixes or improvements.
+
+### (Option C) Design and implement Pluggable Validator mechanism for SDK
+
+Instead of we add the OpenshiftValidator to [operator-framework/api][oper-api] we would only provide it via the custom "downstream plugin" and make it available only for SDK binary.
+
+**Pros**
+
+- Address an RFE which allow others create their own validators and use them with SDK.
+- Does not affect the users and Pipeline since they will still able to use the upstream binary and the downstream validator module
+- Address the OLM team concerns about add the validator in the upstream api. Also, address similar concerns, if they exist, for SDK CLI in upstream.
+
+**Cons**
+
+- The Validator cannot be used outside of SDK. So, for example, it cannot be used by [audit proposal][audit-ep] if we decide to not make this solution leverage in SDK CLI (binary).
+- Brings higher effort to implement since requires the design and implementation of a new feature for SDK which we have none other use case/request to use that so far which justify it.
+- Shows that will be harder for us to keep it maintained and/or receives contributions since it will leave only in [ocp-release-operator-sdk][ocp-release-operator-sdk].
+- We will be unable to promote the feature for the users since it will not available by default.
+- It does not allow the re-usability of the validator via importing the code implementation to the other projects as the option A or B.
```

Comment:
That's because it will be in a dynamic language of some sort that is reusable by loading it and parsing/running.

Reply:
The ep was clarified. However, not that the idea suggested is not the same added by your here. Your idea/suggestion was added as another alternative solution. See; https://github.com/openshift/enhancements/pull/661#discussion_r598284681